File: app/src/main/java/com/xjm/xxd/dribbird/api/retrofit/UserApi.java
package com.xjm.xxd.dribbird.api.retrofit;

import com.xjm.xxd.dribbird.model.UserBean;

import io.reactivex.Flowable;
import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * @author : xiaoxiaoda
 *         date: 16-12-7
 *         email: daque@hustunique.com
 */
public interface UserApi {

    @GET("user")
    Observable<UserBean> getAuthenticatedUser();

    @GET("users/{userId}")
    Observable<UserBean> getUserInfo(@Path("userId") String userId);

}

File: app/src/main/java/com/xjm/xxd/dribbird/api/retrofit/UserApi.kt
package com.xjm.xxd.dribbird.api.retrofit

import com.xjm.xxd.dribbird.model.UserBean

import io.reactivex.Flowable
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * @author : xiaoxiaoda
 * date: 16-12-7
 * email: daque@hustunique.com
 */
interface UserApi {

    @get:GET("user")
    val authenticatedUser: Observable<UserBean>

    @GET("users/{userId}")
    fun getUserInfo(@Path("userId") userId: String): Observable<UserBean>

}
